python/pyiceberg/table/__init__.py

```diff
@@ -78,7 +80,6 @@ def name(self) -> Identifier:
     def scan(
         self,
         row_filter: Optional[BooleanExpression] = None,
-        partition_filter: Optional[BooleanExpression] = None,
```

Comment:
I was using this to test a fix for slow job planning with PyArrow, so I thought I'd go ahead and include it in this PR.

Reply:
Here's the fix: https://github.com/apache/iceberg/pull/6283